arbnode/broadcast_sync_checker_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021-2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package arbnode
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/offchainlabs/nitro/arbutil"
+)
+
+// Mock SyncStatusProvider for testing
+type mockSyncStatusProvider struct {
+	synced bool
+}
+
+func (m *mockSyncStatusProvider) Synced() bool {
+	return m.synced
+}
+
+// Mock BatchInfoProvider for testing
+type mockBatchInfoProvider struct {
+	batchCount uint64
+	batchMetas map[uint64]BatchMetadata
+	err        error
+}
+
+func (m *mockBatchInfoProvider) GetBatchCount() (uint64, error) {
+	return m.batchCount, m.err
+}
+
+func (m *mockBatchInfoProvider) GetBatchMetadata(seqNum uint64) (BatchMetadata, error) {
+	if m.err != nil {
+		return BatchMetadata{}, m.err
+	}
+	meta, exists := m.batchMetas[seqNum]
+	if !exists {
+		return BatchMetadata{MessageCount: 0}, nil
+	}
+	return meta, nil
+}
+
+func TestBroadcastSyncChecker(t *testing.T) {
+	tests := []struct {
+		name              string
+		synced            bool
+		firstMsgIdx       arbutil.MessageIndex
+		msgCount          int
+		batchCount        uint64
+		thresholdMsgCount arbutil.MessageIndex
+		batchInfoError    bool
+		expected          bool
+		description       string
+	}{
+		{
+			name:        "empty batch - never broadcast",
+			msgCount:    0,
+			expected:    false,
+			description: "Empty message batches should never be broadcast",
+		},
```

Comment:
ShouldBroadcast could be returning true due to batchCount being equal to zero too.
How about using the same parameters used by "not synced - last message at threshold - broadcast" case, but setting batchInfoError to true instead?